dashboard/src/components/OperatorInstance/OperatorInstance.v2.tsx

```diff
@@ -0,0 +1,261 @@
+          result.statefulsets.push(fromCRD(r, cluster, namespace, ownerRef));
+          break;
+        case "DaemonSet":
+          result.daemonsets.push(fromCRD(r, cluster, namespace, ownerRef));
+          break;
+        case "Service":
+          result.services.push(fromCRD(r, cluster, namespace, ownerRef));
+          break;
+        case "Ingress":
+          result.ingresses.push(fromCRD(r, cluster, namespace, ownerRef));
+          break;
+        case "Secret":
+          result.secrets.push(fromCRD(r, cluster, namespace, ownerRef));
+          break;
+        default:
+          result.otherResources.push(fromCRD(r, cluster, namespace, ownerRef));
+      }
+    });
+  } else {
+    const emptyCRD = { kind: "", name: "", version: "" };
+    // The CRD definition doesn't define any service so pull everything
+    result = {
+      deployments: [fromCRD({ ...emptyCRD, kind: "Deployment" }, cluster, namespace, ownerRef)],
+      ingresses: [fromCRD({ ...emptyCRD, kind: "Ingress" }, cluster, namespace, ownerRef)],
+      statefulsets: [fromCRD({ ...emptyCRD, kind: "StatefulSet" }, cluster, namespace, ownerRef)],
+      daemonsets: [fromCRD({ ...emptyCRD, kind: "DaemonSet" }, cluster, namespace, ownerRef)],
+      services: [fromCRD({ ...emptyCRD, kind: "Service" }, cluster, namespace, ownerRef)],
+      secrets: [fromCRD({ ...emptyCRD, kind: "Secret" }, cluster, namespace, ownerRef)],
+      otherResources: [],
+    };
+  }
+  return result;
+}
+
+function OperatorInstance({
+  cluster,
+  namespace,
+  csvName,
+  crdName,
+  instanceName,
+}: IOperatorInstanceProps) {
+  const dispatch: ThunkDispatch<IStoreState, null, Action> = useDispatch();
+  const [crd, setCRD] = useState(undefined as IClusterServiceVersionCRD | undefined);
+  const [icon, setIcon] = useState(placeholder);
+  const [deleting, setDeleting] = useState(false);
+  const [resourceRefs, setResourceRefs] = useState({
+    ingresses: [],
+    deployments: [],
+    statefulsets: [],
+    daemonsets: [],
+    otherResources: [],
+    services: [],
+    secrets: [],
+  } as IAppViewResourceRefs);
+  const {
+    services,
+    ingresses,
+    deployments,
+    statefulsets,
+    daemonsets,
+    secrets,
+    otherResources,
+  } = resourceRefs;
+  const [modalIsOpen, setModalIsOpen] = useState(false);
+  const closeModal = () => setModalIsOpen(false);
+  const openModal = () => setModalIsOpen(true);
+
+  const {
+    operators: {
+      isFetching,
+      csv,
+      resource,
+      errors: { resource: errors },
+    },
+    config: { kubeappsCluster },
+  } = useSelector((state: IStoreState) => state);
+
+  useEffect(() => {
+    dispatch(actions.operators.getResource(namespace, csvName, crdName, instanceName));
+    dispatch(actions.operators.getCSV(namespace, csvName));
+  }, [dispatch, namespace, csvName, crdName, instanceName]);
+
+  useEffect(() => {
+    if (csv) {
+      parseCSV(csv, crdName, setIcon, setCRD);
+    }
+  }, [csv, crdName]);
```

Comment:
Looks like it's missing deps?

Reply:
nope, the only vars are `csv` and `crdName`, the rest are functions. Just FYI, there is a linter check that verifies this so it's not possible to specify less variables than needed.